cmake/EnableCompilerWarnings.cmake

```diff
@@ -92,6 +92,7 @@ else()
                 -Wno-unused-command-line-argument
                 -Wno-weak-vtables
                 -Wno-covered-switch-default
+                -Wno-unused-result
```

Comment:
Because its beyond the scope of this PR. This PR focuses on enabling CK in MIOpen. We plan to properly fix these warnings in subsequent PRs

Reply:
> This PR focuses on enabling CK in MIOpen.

Regardless of the task needed, the warnings should be required to be fixed(or suppressed locally with pragmas) for all PRs.

> We plan to properly fix these warnings in subsequent PRs

If its not being fixed when driven by features/defect then it will less likely be fixed by aesthetics.

Also, CK should wait until MIOpen has C++17 fully enabled before updating their headers to use C++17.